plugins/mac-videotoolbox/encoder.c

```diff
@@ -447,14 +543,16 @@ static void dump_encoder_info(struct vt_encoder *enc)
 		"\trc_max_bitrate:        %d (kbps)\n"
 		"\trc_max_bitrate_window: %f (s)\n"
 		"\thw_enc:                %s\n"
-		"\tprofile:               %s\n",
+		"\tprofile:               %s\n"
+		"\tcodec_type:            %.4s\n",
 		enc->vt_encoder_id, enc->rate_control, enc->bitrate,
 		enc->quality, enc->fps_num, enc->fps_den, enc->width,
 		enc->height, enc->keyint, enc->limit_bitrate ? "on" : "off",
 		enc->rc_max_bitrate, enc->rc_max_bitrate_window,
 		enc->hw_enc ? "on" : "off",
 		(enc->profile != NULL && !!strlen(enc->profile)) ? enc->profile
-								 : "default");
+								 : "default",
+		(char *)&enc->codec_type);
```

Comment:
This appears to be printing the values the wrong way round (`ncpa` instead of `apcn`, `1cvh` instead of `hvc1`, etc).
Could it use `codec_type_to_print_fmt` instead? That would also print `h264` instead of `avc1` (and `hevc` instead of `hvc1`), making it simpler for people looking at the log to understand.

Reply:
That's because `CMVideoCodecType` is a `FourCharCode`, represented as a packed `UInt32` which (like `OSCode`) comes from the dark days of PowerPC and hence is Big Endian. We'd either need to construct a string from the bytes in reverse order or just use the `codec_type_to_print_fmt` helper function that already exists for that purpose.

Reply:
Looks like it's fixed now.